CHANGELOG.md

```diff
@@ -6,6 +6,7 @@ All notable changes to this project will be documented in this file. From versio
 
 ### Added
 
+- Use `pg_basetype()` for domain type resolution on PostgreSQL 17+ by @joelonsql in #XXXX
```

Comment:
What's the observable effect, if any, for the user? Is this faster? If so, we should say that. I would not consider this a new "feature". It's likely either a performance improvement or a refactor. Although as a refactor it'd only increase complexity for now.

TLDR: If this doesn't make things faster, we should wait until we drop support for PG 16 - *then* this will make the code less complex.

Reply:
Yeah, I know it was added by Steve. I still think that if we can't show a performance improvement, we should rather wait until the function is available in all supported PG versions, because at this stage it does not simplify our code, yet.